### PR TITLE
cleanup: Fastboard

### PR DESCRIPTION
**Pull request by sethtroisi**
simplify some fastboard logic

add `m_squaresize = m_boardsize + 2;` which tracks the size of the underlying board representation 
add `white_to_move()` to complicment `black_to_move()`
utilize `white_to_move()` and `black_to_move()` in a number of places

### CODE REVIEW

**Comment by gcp** [comment]
This has AutoGTP changes intermixed.

**Comment by sethtroisi** [comment]
rebased.

File: src/FastBoard.cpp [hunk 1]
```diff
@@ -616,14 +611,19 @@ int FastBoard::get_prisoners(int side) {
...
+}
+
+bool FastBoard::black_to_move() const {
     return m_tomove == BLACK;
 }
 
-int FastBoard::get_to_move() {
-    return m_tomove;
+bool FastBoard::white_to_move() const {
+    return m_tomove == BLACK;
```
Comment (gcp) [inline src/FastBoard.cpp]: This looks wrong :P

**Comment by sethtroisi** [comment]
merged next
fixed white_to_move
verified by comparing eval of empty board in this branch and next

File: src/Network.cpp [hunk 2]
```diff
@@ -636,9 +636,8 @@ void Network::gather_features(GameState * state, NNPlanes & planes) {
     BoardPlane& black_to_move  = planes[16];
     BoardPlane& white_to_move  = planes[17];
 
-    int to_move = state->get_to_move();
-    bool whites_move = to_move == FastBoard::WHITE;
-    if (whites_move) {
+    auto to_move = state->get_to_move();
+    if (to_move == FastBoard::WHITE) {
```
Comment (gcp) [inline src/Network.cpp]: state->board.white_to_move()?

Comment (sethtroisi) [reply src/Network.cpp]: yes, this is even cleaner.

**Comment by sethtroisi** [comment]
ready for a 2nd look

**Pull request merged**